Ray.jl/src/runtime_env.jl

```diff
@@ -1,15 +1,80 @@
 struct RuntimeEnv
     project::String
+    package_imports::Expr
+
+    function RuntimeEnv(project, package_imports)
+        return new(project, process_import_statements(package_imports))
+    end
 end
 
-RuntimeEnv(; project=project_dir()) = RuntimeEnv(project)
+function RuntimeEnv(; project=project_dir(), package_imports=Expr(:block))
+    return RuntimeEnv(project, package_imports)
+end
 
 function json_dict(runtime_env::RuntimeEnv)
+    env_vars = Dict("JULIA_PROJECT" => runtime_env.project)
```

Comment:
probably good to at least add a TODO to support user-provided `env_vars` in the runtime env too.